pkg/storageops/storageops_test.go

```diff
@@ -0,0 +1,152 @@
+package storageops_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/opsworks"
+	"github.com/libopenstorage/openstorage/pkg/storageops"
+	"github.com/libopenstorage/openstorage/pkg/storageops/aws"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	newDiskPrefix      = "openstorage-test"
+	newDiskDescription = "Disk created by Openstorage tests"
+	newDiskSizeInGB    = 10
+)
+
+var drivers map[string]storageops.Ops
+var diskTemplates map[string]map[string]interface{}
+var diskName = fmt.Sprintf("%s-%s", newDiskPrefix, uuid.NewV4())
+
+func initDrivers(t *testing.T) {
+	drivers = make(map[string]storageops.Ops)
+	diskTemplates = make(map[string]map[string]interface{})
+
+	// AWS
+	if d, err := aws.NewEnvClient(); err != aws.ErrAWSEnvNotAvailable {
```

Comment:
nit: we could execute this AWS specific code under the aws/storageops_test.go and then pass in the StorageOps interface to this test suite. In this way this file would be cloud provider agnostic.

Reply:
Changed API in incremental